heaps/heap_sort.py

```diff
@@ -1,8 +1,20 @@
-
+from heaps.min_heap import MinHeap
 
 def heap_sort(list):
     """ This method uses a heap to sort an array.
-        Time Complexity:  ?
-        Space Complexity: ?
+        Time Complexity:  (log n)
+        Space Complexity: (O n)
     """
-    pass
\ No newline at end of file
+
+    heap = MinHeap()
+
+    for i in list:
+        heap.add(i)
+
+    i = 0
+
+    while not heap.empty():
+        list[i] = heap.remove()
+        i += 1
```

Comment:
Note the since this isn't a fully in-place solution (the MinHeap has a O(n) internal store), we don't necessarily need to modify the passed in list. The tests are written to check the return value, so we could unpack the heap into a new result list to avoid mutating the input.

```py
    result = []
    while not heap.empty():
        result.append(heap.remove())

    return result
```